src/otx/v2/adapters/torch/mmengine/mmdet/modules/evaluation/evaluator.py

```diff
@@ -0,0 +1,469 @@
+            gt_masks.append(gt_per_classes)
+        return gt_masks
+
+    def compute_metrics(
+        self,
+        results: list,
+    ) -> OrderedDict[str, float]:
+        """Evaluate the dataset.
+
+        Args:
+            results (list): Testing results of the dataset.
+        """
+        logger: MMLogger = MMLogger.get_current_instance()
+        eval_results = OrderedDict()
+        gts, preds = zip(*results)
+        mean_aps = []
+        for iou_thr in self.iou_thrs:  # pylint: disable=redefined-argument-from-local
+            print_log(f'\n{"-" * 15}iou_thr: {iou_thr}{"-" * 15}')
+            if self.dataset_meta["domain"] == Domain.DETECTION:
+                mean_ap, _ = eval_map(
+                    preds,
+                    gts,
+                    scale_ranges=self.scale_ranges,
+                    iou_thr=iou_thr,
+                    dataset=self.dataset_meta["classes"],
+                    logger=logger,
+                    eval_mode=self.eval_mode,
+                    use_legacy_coordinate=True,
+                )
+            else:
+                mean_ap, _ = self.evaluate_mask(gts, preds, logger, iou_thr)
+            mean_aps.append(mean_ap)
+            eval_results[f"AP{int(iou_thr * 100):02d}"] = round(mean_ap, 3)
+        eval_results["mAP"] = sum(mean_aps) / len(mean_aps)
+        return eval_results
+
+    def process(self, data_batch: dict, data_samples: Sequence[dict]) -> None:  # noqa: ARG002
+        """Process one batch of data samples and predictions.
+
+        The processed results should be stored in ``self.results``, which will be used to
+        compute the metrics when all batches have been processed.
+
+        Args:
+            data_batch (dict): A batch of data from the dataloader.
+            data_samples (Sequence[dict]): A batch of data samples that
+                contain annotations and predictions.
+        """
+        for data_sample in data_samples:
+            gt = copy.deepcopy(data_sample)
+            gt_instances = gt["gt_instances"]
+            gt_ignore_instances = gt["ignored_instances"]
+            ann = {
+                "labels": gt_instances["labels"].cpu().numpy(),
+                "bboxes": gt_instances["bboxes"].cpu().numpy(),
+                "bboxes_ignore": gt_ignore_instances["bboxes"].cpu().numpy(),
+                "labels_ignore": gt_ignore_instances["labels"].cpu().numpy(),
+            }
+
+            pred = data_sample["pred_instances"]
+            pred_bboxes = pred["bboxes"].cpu().numpy()
+            pred_scores = pred["scores"].cpu().numpy()
+            pred_labels = pred["labels"].cpu().numpy()
+
+            if self.dataset_meta["domain"] == Domain.INSTANCE_SEGMENTATION:
+                ann["masks"] = gt_instances["masks"]
+                ann["masks_ignore"] = gt_ignore_instances["masks"]
+                pred_masks = np.array([encode_mask_results(mask)[0] for mask in pred["masks"]])
+
+            dets = []
```

Comment:
same here full word looks much better than abbreviation. Not sure why MMLab prefers abbreviations

Reply:
Maybe it's just a personal preference...